.github/workflows/ci.yaml

```diff
@@ -0,0 +1,86 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches: [master]
+  workflow_dispatch:
+    inputs:
+      debug_enabled:
+        description: "Run the build with tmate debugging enabled"
+        required: false
+jobs:
+  tox:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        tox-env: [lint, docs, check-packaging]
+    steps:
+      - name: Cancel previous workflows that are still running
+        uses: styfle/cancel-workflow-action@0.8.0
+        with:
+          access_token: ${{ github.token }}
+      - name: Checkout latest commit
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # fetch all history with version tags
+      - name: Set up python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.7"
+      - name: Set up pip cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ matrix.tox-env }}-${{ runner.os }}-${{ hashFiles('tox.ini') }}
```

Comment:
Is it useful? Indeed, with the current key, the cache will never be reused because `matrix.tox-env` is unique. However, I am new to https://github.com/actions/cache and I might have missed something in the documentation.

Reply:
I think the idea is not to share the cache between the jobs, but to re-use it the next time the same job is run.

Reply:
I see. Resolving.

Note: We need then to remember this on long running pull requests. Indeed, we might have older package versions on such pull request compared to a more recent one, while the CI runs are done around the same time.

Reply:
Yes, good point. Would probably good to clear the cache if we change our dependencies drastically.